pkg/apis/networking/v1alpha1/ingress_validation.go

```diff
@@ -166,3 +167,14 @@ func (t *IngressTLS) Validate(ctx context.Context) *apis.FieldError {
 	}
 	return all
 }
+
+// Validate inspects and validates HTTPOption object.
+func (t HTTPOption) Validate(ctx context.Context) (all *apis.FieldError) {
+	switch t {
+	case HTTPOptionEnabled, HTTPOptionDisabled, HTTPOptionRedirected:
+		break
+	default:
+		all = all.Also(apis.ErrInvalidValue(t, "httpOption"))
+	}
+	return all
+}
```

Comment:
We definitely cannot add this, it means that ALL existing kingress are invalid.

Reply:
Wouldn't defaulting solve this?

Reply:
Defaulting only solves it if we change something in spec.  In the meantime, I believe status updates will fail to validate.  Regardless, we cannot default (yet) because it violates our ability to downgrade.

Reply:
Would it be OK to allow empty value of HTTPOption? I think if we allow empty string, then the Validate function should be fine, right?

Reply:
Yeah, if it's there, we should validate it, but we shouldn't require it to be there.